### ISSUE TITLE
Support for Groth16 (without GT)

### ISSUE THREAD

**Issue by HarryR** [EYBlockchain/zksnark-verifier-standard#2]
See:

  * https://github.com/HarryR/ethsnarks/tree/master/src/r1cs_gg_ppzksnark_zok
  * https://github.com/Zoffee/zoffee-onchain/blob/master/contracts/Verifier_Groth16sansGT.sol

Reduces on-chain verification cost to about 450k gas, requires circuit-level anti-malleability checks.

**Comment by MichaelConnorOfficial** [comment]
Please refer to my reply re GM17 - I’m not sure I’m clear on how or why ‘support for’ each different proving method should be addressed through the interface of a standard. It feels like these cases should be handled within the body of the functions of people’s implementations of these standards. What do you think?

**Comment by HarryR** [comment]
Yup the interface should be completely agnostic to the underlying implementation.

I was merely making notes for the other two verifier implementations which are currently in-use.
